cpp/src/command_classes/SceneActuatorConf.cpp

```diff
@@ -0,0 +1,164 @@
+//
+//-----------------------------------------------------------------------------
+
+#include "command_classes/CommandClasses.h"
+#include "command_classes/SceneActuatorConf.h"
+#include "Defs.h"
+#include "Msg.h"
+#include "Node.h"
+#include "Driver.h"
+#include "platform/Log.h"
+
+using namespace OpenZWave;
+
+enum SceneActuatorConfCmd
+{
+	SceneActuatorConfCmd_Set 	= 0x01,
+	SceneActuatorConfCmd_Get 	= 0x02,
+	SceneActuatorConfCmd_Report = 0x03
+};
+
+
+//-----------------------------------------------------------------------------
+// <SceneActuatorConf::RequestState>
+// Nothing to do for SceneActuatorConf
+//-----------------------------------------------------------------------------
+bool SceneActuatorConf::RequestState
+(
+	uint32 const _requestFlags,
+	uint8 const _instance,
+	Driver::MsgQueue const _queue
+)
+{
+	if( ( _requestFlags & RequestFlag_Dynamic ) )
+	{
+//		RequestValue( _requestFlags, 0x01, _instance, _queue );
+//		SetScene( 0x00, 0xff, 0xff );
+//		RequestValue( _requestFlags, 0x01, _instance, _queue );
+//		RequestValue( _requestFlags, 0x02, _instance, _queue );
+//		RequestValue( _requestFlags, 0x03, _instance, _queue );
+//		RequestValue( _requestFlags, 0x04, _instance, _queue );
+		return RequestValue( _requestFlags, 0, _instance, _queue ); // request active scene
+	}
+
+	return false;
+}
+
+//-----------------------------------------------------------------------------
+// <SceneActuatorConf::RequestValue>
+// Nothing to do for Association
+//-----------------------------------------------------------------------------
+bool SceneActuatorConf::RequestValue
+(
+	uint32 const _requestFlags,
+	uint8 const _sceneId,
+	uint8 const _instance,
+	Driver::MsgQueue const _queue
+)
+{
+	if( _instance != 1 )
+	{
+		// This command class doesn't work with multiple instances
+		return false;
+	}
+
+	Msg* msg = new Msg( "SceneActuatorConfCmd_Get", GetNodeId(), REQUEST, FUNC_ID_ZW_SEND_DATA, true, true, FUNC_ID_APPLICATION_COMMAND_HANDLER, GetCommandClassId() );
+	msg->Append( GetNodeId() );
+	msg->Append( 3 );
+	msg->Append( GetCommandClassId() );
+	msg->Append( SceneActuatorConfCmd_Get );
+	msg->Append( _sceneId );
+	msg->Append( GetDriver()->GetTransmitOptions() );
+	GetDriver()->SendMsg( msg, _queue );
+	return true;
+}
+
+
+void SceneActuatorConf::SetScene
+(
+	uint8 const _sceneId,
+	uint8 const _level,
+	uint8 const _duration
+)
+{
+	// once a actuator has activated the scene it can only be undone by a reset
+	// TODO _bReplyRequired is false else we get a timeout. Why is this needed?
+	Msg* msg = new Msg( "SceneActuatorConfCmd_Set", GetNodeId(), REQUEST, FUNC_ID_ZW_SEND_DATA, true, false, FUNC_ID_APPLICATION_COMMAND_HANDLER, GetCommandClassId() );
+	msg->Append( GetNodeId() );
+	msg->Append( 6 );
+	msg->Append( GetCommandClassId() );
+	msg->Append( SceneActuatorConfCmd_Set );
+	msg->Append( _sceneId );
+	msg->Append( _duration );
+	msg->Append( 0x80 ); // use our value not the current nodes level
```

Comment:
Yeah - But broadcast would mean that all nodes that support this CC would save this scene. I'm not sure if thats the right option. 

Or - Like some classes, Set doesn't require a reply. In that case, you can drop all the rest of the variables after the false.